CMakeLists.txt

```diff
@@ -134,7 +147,12 @@ install(FILES "$<TARGET_FILE:mimalloc>" DESTINATION lib)  # duplicate the .so in
 
 # single object file for more predictable static overriding
 add_library(mimalloc-obj OBJECT src/static.c)
-target_compile_definitions(mimalloc-obj PRIVATE ${mi_defines} MI_MALLOC_OVERRIDE)
+target_compile_definitions(mimalloc-obj PRIVATE ${mi_defines})
+if(NOT WIN32)
+  # It is only possible to override malloc on Windows when building as a DLL.
+  # (src/alloc-override.c)
+  target_compile_definitions(mimalloc-obj PRIVATE $<$<BOOL:${OVERRIDE}>:MI_MALLOC_OVERRIDE>)
```

Comment:
Here, too, I think it would make sense to just insert this block instead:

```cmake
if(OVERRIDE MATCHES "ON")
  list(APPEND mi_defines MI_MALLOC_OVERRIDE)
endif()
```

It is easier to read, as it imitates the rather idiomatic (and simple) CMake style used in the rest of the file. This is _particularly_ important when one keeps in mind that most developers do not really want to become CMake experts.